### ISSUE TITLE
reduce_into false positive for Combine usage

### ISSUE THREAD

**Issue by jshier** [realm/SwiftLint#3244]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

SwiftLint's current `reduce_into` rule is triggered when using Combine's `reduce(_:_:)` operator. There is no "into" version of this operator.

For example, when operating on many publishers of arrays:

```swift
Publishers.MergeMany(arrayPublishers)
            .reduce([]) { $0 + $1 }
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? **0.44.13**
* Installation method used (Homebrew, CocoaPods, building from source, etc)? CocoaPods
* Which Xcode version are you using (check `xcodebuild -version`)? 11.5
* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

TBD

**Comment by gerlv** [comment]
Just run into this issue.

Is there a workaround other than disabling this rule?
